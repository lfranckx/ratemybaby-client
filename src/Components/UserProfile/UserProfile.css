#profilepic {
    height: 250px;
    width: 250px;
    border-radius: 50%;
    background-color: white;
}

.about {
    text-align: left;
    border-radius: 4px;
    padding: 5px;
    margin-top: 20px;
}

#continue {
    position: relative;
    top: 10px;
}
