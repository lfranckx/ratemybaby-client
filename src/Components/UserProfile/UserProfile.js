import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import HamburgerContext from '../../Contexts/HamburgerContext'
import './UserProfile.css'

export default class UserProfile extends Component {

    static contextType = HamburgerContext

    render() {
        const { baby } = this.props     
        if (!baby) {
            return <div className='loading'>Loading...</div>
        }

        let rating = (baby.total_score / baby.total_votes)
        let percent = rating * 100
        let roundPercent = Math.round(percent) + '%'

        return (
            <>
            <section className="profile-container">
                <img src={baby.image_url} alt="profile" className="baby-pic" />
                <Link id="change-photo" to={`/uploadimage/${baby.id}`}>Change Photo</Link>  
                <div className="name-age">
                    <h3 id="user-baby-name">{baby.baby_name}</h3>
                    <span className="age">
                         <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/bdaycake.png" 
                            alt="birthday-cake" 
                            id="cake" />
                        {baby.age} {baby.age_format}
                    </span>
                    <span className="country">
                        <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/house.png" 
                            alt="house" 
                            id="house" />
                        {baby.country}
                    </span>
                </div>
                <div className="about">
                    {baby.about}
                </div>
                <div className='edit-info-box'>
                    <div className="user-rating">{roundPercent}</div>
                    <button className="edit-info">
                        <Link to={`/editprofile/${baby.id}`}>Edit Info</Link>
                    </button>
                </div>
                <Link to={`/delete/${baby.id}`}>Delete</Link> 
            </section>
            </>
        )
    }
}