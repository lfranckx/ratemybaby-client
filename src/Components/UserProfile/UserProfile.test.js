import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserProfile from './UserProfile';

const context = {
    baby: {
        "baby_name": "Test",
        "age": "7", 
        "age_format": "months",
        "country": "United States",
        "about": "Testing data",
        "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
        "total_score": "25",
        "total_votes": "35",
        "parent_id": "2"
    }
}

describe(`UserProfile component`, () => {
    it(`renders loading by default`, () => {
        const wrapper = shallow(<UserProfile />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it(`renders a profile given context`, () => {
        const wrapper = shallow(
            <UserProfile /> 
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});