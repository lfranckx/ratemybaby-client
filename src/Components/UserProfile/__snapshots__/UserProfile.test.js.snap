// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component renders a profile given props 1`] = `
<Fragment>
  <section
    className="profile-container"
  >
    <img
      alt="profile"
      className="baby-pic"
      src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg"
    />
    <Link
      id="change-photo"
      to="/uploadimage/undefined"
    >
      Change Photo
    </Link>
    <div
      className="name-age"
    >
      <h3
        id="user-baby-name"
      >
        Test
      </h3>
      <span
        className="age"
      >
        <img
          alt="birthday-cake"
          id="cake"
          src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/bdaycake.png"
        />
        7
         
        months
      </span>
      <span
        className="country"
      >
        <img
          alt="house"
          id="house"
          src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/house.png"
        />
        United States
      </span>
    </div>
    <div
      className="about"
    >
      Testing data
    </div>
    <div
      className="edit-info-box"
    >
      <div
        className="user-rating"
      >
        71%
      </div>
      <button
        className="edit-info"
      >
        <Link
          to="/editprofile/undefined"
        >
          Edit Info
        </Link>
      </button>
    </div>
    <Link
      to="/delete/undefined"
    >
      Delete
    </Link>
  </section>
</Fragment>
`;

exports[`UserProfile component renders loading by default 1`] = `
<div
  className="loading"
>
  Loading...
</div>
`;
